Use us ticker for tickless on devs with wrapper

c1728p9 opened the pull request:
### Description

The low power ticker wrapper layer adds a large amount of interrupt latency. This can cause dropped bytes at a baud of 115200 on some devices. To prevent this by default use the microsecond ticker for tickless on devices which make use of the low power ticker wrapper.

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

@jeromecoutant 
@ithinuel

c1728p9 committed:
Use us ticker for tickless on devs with wrapper

The low power ticker wrapper layer adds a large amount of interrupt
latency. This can cause dropped bytes at a baud of 115200 on some
devices. To prevent this by default use the microsecond ticker for
tickless on devices which make use of the low power ticker wrapper.

kjbracey commented on rtos/TARGET_CORTEX/mbed_rtx_idle.cpp:
```diff
@@ -96,6 +101,11 @@ extern "C" {
         uint32_t ticks_to_sleep = osKernelSuspend();
         const bool block_deep_sleep = ticks_to_sleep <= MBED_CONF_TARGET_DEEP_SLEEP_LATENCY;
 
+#if MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER
```
Surely just

    const bool block_deep_sleep = MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER ||
                                  ticks_to_sleep <= MBED_CONF_TARGET_DEEP_SLEEP_LATENCY;

c1728p9 replied:
Done

kjbracey commented on rtos/TARGET_CORTEX/mbed_rtx_idle.cpp:
```diff
@@ -47,7 +48,11 @@ extern "C" {
         // Do not use SingletonPtr since this relies on the RTOS
         if (NULL == os_timer) {
-            os_timer = new (os_timer_data) rtos::internal::SysTimer();
+#if MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER
```
Should this also be automatically inferred if `DEVICE_LPTICKER` is unset, rather than needing to be set manually?

    #if !DEVICE_LPTICKER
    #  define USE_US_TICKER true
    #else 
    #  define USE_US_TICKER MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER
    #endif

maybe.

c1728p9 replied:
If it is automatically inferred then `MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER` may be false (default) yet the microsecond ticker is used. This could lead to problems if `MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER` is used elsewhere. I added compiler asserts so you get a clear error message if the required DEVICE_*TICKER is missing.

c1728p9 committed:
Apply astyle fixes

Apply the fixes found from astyle.

c1728p9 committed:
Use the RTC for the low power ticker on L073RZ

Due to a bug the LPTIM peripheral can cause interrupts to fire at the
wrong time on the NUCLEO_L073RZ. This patch switches the backend
of the low power ticker to the RTC until this bug is addressed.

c1728p9 committed:
Skip deep sleep test when running from US ticker

Skip the systimer deep sleep test when running from the microsecond
ticker, since the microsecond ticker doesn't support operation in deep
sleep mode.
